src/main/java/org/jenkinsci/plugins/mesos/OfferQueue.java

```diff
@@ -0,0 +1,121 @@
+package org.jenkinsci.plugins.mesos;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.mesos.Protos;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * This class acts as a buffer of Offers from Mesos.  By default it holds a maximum of 100 Offers.
+ */
+public class OfferQueue {
+    private final Logger logger = Logger.getLogger(JenkinsScheduler.class.getName());
+
+    private static final int DEFAULT_CAPACITY = 100;
```

Comment:
We haven't allowed it's configuration through any means in the SDK.  It has proven a good value at the limits of Mesos offer scale.  Our plan was to make it configurable if it ever becomes necessary.

Generally we tend to _not_ allow configuration of things by default to avoid the 300 nobs to turn scenario.